src/components/text/text.styles.ts

```diff
@@ -0,0 +1,339 @@
+    }
+
+    h6 {
+      font-size: ${headings.h6.fontSize};
+      line-height: ${headings.h6.lineHeight};
+    }
+    h6:not(:last-child) {
+      ${logicalCSS('margin-bottom', headingMarginBottom(headings.h6.fontSize))}
+    }
+    * + h6 {
+      ${logicalCSS('margin-top', headingMarginTop(headings.h6.fontSize))}
+    }
+
+    p,
+    ul,
+    ol,
+    dl,
+    blockquote,
+    img,
+    pre {
+      ${logicalCSS('margin-bottom', lineHeightSize)}
+    }
+
+    ul,
+    ol {
+      ${logicalCSS('margin-left', lineHeightSize)}
+    }
+  
+    blockquote {
+      font-size: ${fontSize};
+      padding: ${lineHeightSize};
+    }
+
+    dd + dt {
+      ${logicalCSS('margin-top', fontSize)}
+    }
+
+    dt,
+    .eui-definitionListReverse dd {
+      font-size: ${fontSize};
+      line-height: ${lineHeight};
+    }
+
+    .eui-definitionListReverse dt {
+      font-size: ${euiFontSize('xs', euiTheme, options).fontSize};
+      color: ${euiTheme.colors.text};
+    }
+
+    small {
+      font-size: ${euiFontSize('s', euiTheme, options).fontSize};
+    }
+
+    pre {
+      padding: ${fontSize};
+    }
+
+    code {
+      font-size: .9em; // 90% of parent font size
+    }
+  `;
+};
+
+/**
+ * Styles
+ */
+export const euiTextStyles = ({ euiTheme }: UseEuiTheme) => ({
+  euiText: css`
+    ${euiText(euiTheme, true)};
+
+    // EuiImage with floats are often used within EuiText.
+    clear: both;
+
+    // Style anchors that don't have a class. This prevents overwriting "buttons"
+    // and other stylized elements passed in.
+    a:not([class]) {
+      ${euiLinkCSS(euiTheme)}
+    }
+
+    img {
+      display: block;
+      width: 100%;
+    }
+
+    ul {
+      list-style: disc;
+    }
+
+    ol {
+      list-style: decimal;
+    }
+
+    blockquote:not(.euiMarkdownFormat__blockquote) {
+      position: relative;
+      text-align: center;
+      margin-left: auto;
+      margin-right: auto;
+      font-family: Georgia, Times, Times New Roman, serif;
+      font-style: italic;
+      letter-spacing: normal;
+
+      p:last-child {
+        margin-bottom: 0;
+      }
+
+      &:before,
+      &:after {
+        position: absolute;
+        content: '';
+        height: 2px;
+        width: 50%;
+        right: 0;
+        transform: translateX(-50%);
+        background: ${euiTheme.colors.darkShade};
+      }
+
+      &:before {
+        top: 0;
+      }
+
+      &:after {
+        bottom: 0;
+      }
+    }
+
+    h1 {
+      ${euiTitle('l', euiTheme)}
+    }
+
+    h2 {
+      ${euiTitle('m', euiTheme)}
+    }
+
+    h3 {
+      ${euiTitle('s', euiTheme)}
+    }
+
+    h4,
+    dt {
+      ${euiTitle('xs', euiTheme)}
+    }
+
+    h5 {
+      ${euiTitle('xxs', euiTheme)}
+    }
+
+    h6 {
+      ${euiTitle('xxxs', euiTheme)}
+      text-transform: uppercase;
+    }
+
+    h1,
+    h2,
+    h3,
+    h4,
+    h5,
+    h6,
+    dt {
+      color: inherit;
+    }
+
+    pre {
```

Comment:
We should also really de-scope this so we're not duplicating EuiCodeBlock styles.
```suggestion
    pre:not(.euiCodeBlock__pre) {
```